files/system/usr/bin/flatpak-set-perms.sh

```diff
@@ -0,0 +1,267 @@
+#!/usr/bin/env bash
+
+# Ivo Damjanovic 2024
+
+set -oue pipefail
+
+# Function to display usage information
+usage() {
+    echo "Set Flatpak Overrides. Usage: $0 [-h] [-s] [-u] [-t] [-f] [-b]"
+    echo "-s and -u set the recommended flatpak permissions"
+    echo "-f and -b allow flatpak and backup tools to work"
+    echo "-t allows custom themes to be used within flatpaks"
+    echo "Options:"
+    echo "  -h   Option H: Show usage menu"
+    echo "  -s   Option S: Set System Overrides"
+    echo "  -u   Option U: Set User Overrides"
+    echo "  -t   Option T: Set Theme Overrides for custom theme support"
+    echo "  -f   Option F: Set Flatpak tools Overrides"
+    echo "  -b   Option B: Set Backup tools Overrides"
+    exit 0
+}
+
+# Function for User/System Overrides
+set_override() {
+    local scope="$1"  # 'user' or 'system'
+    local flatpak_command="flatpak override --$scope"
+    
+    # the base override itself, we use the same override for system and user
+    # https://docs.flatpak.org/en/latest/sandbox-permissions.html
+    # we disallow access to filesystem host and home, allow partly access to host read-only
+    # we prefer the wayland socket and only allow x11 as fallback
+    # we disallow access to IPC and Devel
+    # we disallow bus socket access
+    $flatpak_command \
```

Comment:
as already discussed on discord, users who don't know to check their app permissions are the same users who would blindly turn permissions on if something isn't working. As such, blanket globally disabling permissions isn't a good approach.